docs/source/command_line.rst

```diff
@@ -622,6 +622,19 @@ Miscellaneous
     have many scripts that import a large package, the behavior enabled
     by this flag is often more convenient.)
 
+``--new-semantic-analyzer``
+    This flag switches to an improved, experimental implementation of
+    the *semantic analyzer*, which mypy uses internally to bind Python
+    names to definitions. The old and the new semantic analyzers
+    mostly behave identically.  The new semantic analyzer is better at
+    handling import cycles and forward references to definitions. It
+    also fixes inconsistencies between the daemon and non-daemon modes,
+    and it detects additional error conditions.
+
+    The next mypy release will use the new semantic analyzer by
+    default, and the old semantic analyzer will be removed in the next
+    release after that.
```

Comment:
Maybe use the actual expected release numbers instead of "next" and "next after that"?

Reply:
I'd rather not, since we may change our mind about that. However I would like to hedge a bit by adding "Likely, ...".